src/scripts/main.js

```diff
@@ -1 +1,77 @@
 'use strict';
+
+(() => {
+  const body = document.querySelector('body');
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+  });
+
+  const notification = (parentNode, classes, text) => {
+    const node = document.createElement('div');
+
+    node.classList.add(...classes);
+    node.textContent = text;
+    node.dataset.qa = 'notification';
+    parentNode.append(node);
+  };
+
+  const firstPromise = new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
+  });
```

Comment:
This is a potential bug. The 'click' event listener inside the promise will never be removed. This means that if the user clicks on the document after the promise is resolved or rejected, the event will still be triggered. To avoid this, you can use the 'once' option to ensure that the event listener is only invoked once.